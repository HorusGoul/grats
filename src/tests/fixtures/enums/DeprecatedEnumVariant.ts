/** @gqlType */
class SomeType {
  /** @gqlField */
  hello: string;
}

/** @gqlEnum */
enum Enum {
  /**
   * Valid enum value.
   * @deprecated Use something else.
   */
  VALID = "VALID",
  /** Invalid enum value. */
  INVALID = "INVALID",
}
