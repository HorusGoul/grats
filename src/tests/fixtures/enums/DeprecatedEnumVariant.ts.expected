-----------------
INPUT
----------------- 
/** @GQLType */
class Query {
  /** @GQLField */
  hello: string;
}

/** @GQLEnum */
enum Enum {
  /**
   * Valid enum value.
   * @deprecated Use something else.
   */
  VALID = "VALID",
  /** Invalid enum value. */
  INVALID = "INVALID",
}

-----------------
OUTPUT
-----------------
schema {
  query: Query
}

directive @methodName(name: String!) on FIELD_DEFINITION

type Query {
  hello: String
}

enum Enum {
  "Valid enum value."
  VALID @deprecated(reason: "Use something else.")
  "Invalid enum value."
  INVALID
}